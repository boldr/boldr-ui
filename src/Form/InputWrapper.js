import React from 'react';
import styled from 'styled-components';
import Label from './Label';
import Feedback from './Feedback';

/**
 * A wrapper around form input fields that will act as
 * a utility to enable streamlined labeling, styling,
 * and error handling
 */
export const InputWrapper = props => {
  return (
    <InputWrapperContainer ui={props.ui}>
      {props.label && <Label {...props} />}
      {props.children}
      <Feedback {...props} />
    </InputWrapperContainer>
  );
};

export default InputWrapper;

/**
 * When adding a ui prop to the redux form Field it allows us
 * to override the custom theme per form. Make sure you pass
 * ui an object
 */
const InputWrapperContainer = styled.div`
  margin-bottom: 1rem;
`;
