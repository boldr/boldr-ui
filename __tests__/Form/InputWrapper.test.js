import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import InputWrapper from '../../src/Form/InputWrapper';

const mockFn = jest.fn();
const mockProps = {
  ui: { a: '' },
  label: 'abc',
};

test('<InputWrapper /> renders snapshot', () => {
  const wrapper = shallow(<InputWrapper {...mockProps} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
