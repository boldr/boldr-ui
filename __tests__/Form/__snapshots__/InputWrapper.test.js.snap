// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputWrapper /> renders snapshot 1`] = `
<styled.div
  ui={
    Object {
      "a": "",
    }
  }
>
  <Label
    label="abc"
    required={false}
    ui={
      Object {
        "a": "",
      }
    }
  />
  <Feedback
    label="abc"
    ui={
      Object {
        "a": "",
      }
    }
  />
</styled.div>
`;
